src/Mappers/Parameters/TypeHandler.php

```diff
@@ -338,7 +339,17 @@ private function mapType(
         }
         $innerType = $type instanceof Nullable ? $type->getActualType() : $type;
 
-        if ($innerType instanceof Array_ || $innerType instanceof Iterable_ || $innerType instanceof Mixed_) {
+        if (
+            $innerType instanceof Array_
+            || $innerType instanceof Iterable_
+            || $innerType instanceof Mixed_
+            // Try to match generic phpdoc-provided iterables with non-generic return-type-provided iterables
+            // Example: (return type `\ArrayObject`, phpdoc `\ArrayObject<string, TestObject>`)
+            || ($innerType instanceof Object_
+                && $docBlockType instanceof Collection
+                && (string)$innerType->getFqsen() === (string)$docBlockType->getFqsen()
+            )
+        ) {
             // We need to use the docBlockType
             if ($docBlockType === null) {
                 throw CannotMapTypeException::createForMissingPhpDoc($innerType, $reflector, $argumentName);
```

Comment:
phpstan complies about the unexpected type (`Object_`) for `$type` argument.
Fixing this may require to find a wording for exception message.


```
Parameter #1 $type of static method TheCodingMachine\GraphQLite\Mappers\CannotMapTypeException::createForMissingPhpDoc() expects phpDocumentor\Reflection\Types\Array_|phpDocumentor\Reflection\Types\Iterable_|phpDocumentor\Reflection\Types\Mixed_, phpDocumentor\Reflection\Types\Array_|phpDocumentor\Reflection\Types\Iterable_|phpDocumentor\Reflection\Types\Mixed_|phpDocumentor\Reflection\Types\Object_ given.
```


I need suggestions about:
- is it an OK place for the condition?
- should I change the condition to another one?

Thanks